qualtran/bloqs/mean_estimation/mean_estimation_operator.py

```diff
@@ -126,36 +125,36 @@ def decompose_from_registers(
         yield self.select.on_registers(**select_reg)
         yield self.reflect.on_registers(**reflect_reg)
 
-    def _circuit_diagram_info_(self, args: cirq.CircuitDiagramInfoArgs) -> cirq.CircuitDiagramInfo:
-        wire_symbols = [] if self.cv == () else [["(0)", "@"][self.cv[0]]]
-        wire_symbols += ['U_ko'] * (total_bits(self.signature) - total_bits(self.control_registers))
-        return cirq.CircuitDiagramInfo(wire_symbols=wire_symbols)
+    def get_ctrl_system(
+        self, ctrl_spec: Optional['CtrlSpec'] = None
+    ) -> Tuple['Bloq', 'AddControlledT']:
+        if ctrl_spec is None:
+            ctrl_spec = CtrlSpec()
+
+        if self.cv is None and ctrl_spec.shapes in [((),), ((1,),)]:
+            c_encoder = self.code.encoder.controlled(ctrl_spec)
+            assert isinstance(c_encoder, SelectOracle)
+            c_code = attrs.evolve(self.code, encoder=c_encoder)
+            cbloq = attrs.evolve(self, code=c_code, cv=(int(ctrl_spec.cvs[0].item())))
+            (ctrl_reg,) = cbloq.control_registers
+
+            def adder(
+                bb: 'BloqBuilder', ctrl_soqs: Sequence['SoquetT'], in_soqs: dict[str, 'SoquetT']
+            ) -> tuple[Iterable['SoquetT'], Iterable['SoquetT']]:
+                soqs = {ctrl_reg.name: ctrl_soqs[0]} | in_soqs
+                soqs = bb.add_d(cbloq, **soqs)
+                ctrl_soqs = [soqs.pop(ctrl_reg.name)]
+                return ctrl_soqs, soqs.values()
+
+            return cbloq, adder
```

Comment:
you should probably make it so it's possible to factor this out as well, right? we should probably pick what the `control_val`/`cv` argument is called

Reply:
fixed this now, uses the same code as the others